src/mangrove/model.hpp

```diff
@@ -294,22 +301,27 @@ class model {
      * operand, and upsert=true so that objects that aren't already in the collection are
      * automatically inserted.
      *
+     * @param options
+     *      an optional mongocxx::options::update specifying the options to pass to the
+     *      underlying update operation.
+     *
+     * @return the result of the update operation performed in the database.
+     *
      * @see https://docs.mongodb.com/manual/reference/method/db.collection.updateOne/
      */
-    void save() {
+    mongocxx::stdx::optional<mongocxx::result::update> save(
+        mongocxx::options::update options = mongocxx::options::update()) {
         auto id_match_filter = bsoncxx::builder::stream::document{}
                                << "_id" << this->_id << bsoncxx::builder::stream::finalize;
 
         auto update = bsoncxx::builder::stream::document{}
                       << "$set" << boson::to_dotted_notation_document(*static_cast<T*>(this))
                       << bsoncxx::builder::stream::finalize;
 
-        mongocxx::options::update opts{};
+        options.upsert(true);
```

Comment:
If upsert is set to false, it would mean that the object would only be saved if it already exists in the database. I don't really know if that makes sense given the active-record pattern? I would err on the side of keeping it as is, and just documenting that upsert will always be true regardless of what the user sets. thoughts @hanumantmk ?